packages/alfa-style/src/longhands.ts

```diff
@@ -0,0 +1,305 @@
+import FontVariantLigatures from "./property/font-variant-ligatures";
+import FontVariantNumeric from "./property/font-variant-numeric";
+import FontVariantPosition from "./property/font-variant-position";
+import FontWeight from "./property/font-weight";
+import Height from "./property/height";
+import InsetBlockEnd from "./property/inset-block-end";
+import InsetBlockStart from "./property/inset-block-start";
+import InsetInlineEnd from "./property/inset-inline-end";
+import InsetInlineStart from "./property/inset-inline-start";
+import Left from "./property/left";
+import LetterSpacing from "./property/letter-spacing";
+import LineHeight from "./property/line-height";
+import MarginBottom from "./property/margin-bottom";
+import MarginLeft from "./property/margin-left";
+import MarginRight from "./property/margin-right";
+import MarginTop from "./property/margin-top";
+import MinHeight from "./property/min-height";
+import MinWidth from "./property/min-width";
+import Opacity from "./property/opacity";
+import OutlineColor from "./property/outline-color";
+import OutlineOffset from "./property/outline-offset";
+import OutlineStyle from "./property/outline-style";
+import OutlineWidth from "./property/outline-width";
+import OverflowX from "./property/overflow-x";
+import OverflowY from "./property/overflow-y";
+import Position from "./property/position";
+import Right from "./property/right";
+import Rotate from "./property/rotate";
+import TextAlign from "./property/text-align";
+import TextDecorationColor from "./property/text-decoration-color";
+import TextDecorationLine from "./property/text-decoration-line";
+import TextDecorationStyle from "./property/text-decoration-style";
+import TextDecorationThickness from "./property/text-decoration-thickness";
+import TextIndent from "./property/text-indent";
+import TextOverflow from "./property/text-overflow";
+import TextShadow from "./property/text-shadow";
+import TextTransform from "./property/text-transform";
+import Top from "./property/top";
+import Transform from "./property/transform";
+import VerticalAlign from "./property/vertical-align";
+import Visibility from "./property/visibility";
+import WhiteSpace from "./property/white-space";
+import Width from "./property/width";
+import WordSpacing from "./property/word-spacing";
+
+/**
+ * @public
+ */
+export namespace Longhands {
+  export type Property = typeof longHands;
+
+  export type Name = keyof Property;
+
+  /**
+   * Extract the parsed type of a named property.
+   *
+   * @remarks
+   * The parsed type differs from the declared type in that the parsed type
+   * must not include the defaulting keywords as these are handled globally
+   * rather than individually.
+   *
+   * @remarks
+   * The parsed type doesn't really exist in CSS. It is an artefact on how we
+   * handle the default keywords.
+   */
+  export type Parsed<N extends Name> = Property[N] extends Longhand<
+    infer S,
+    infer C
+  >
+    ? S
+    : never;
```

Comment:
**suggestion (non-blocking):** the editor complains about the type variable `C` being unused, it suggests replacing `infer C` with `unknown`
```suggestion
  export type Parsed<N extends Name> = Property[N] extends Longhand<
    infer S,
    unknown
  >
    ? S
    : never;
```